### PR TITLE
[freeglut] updated to 3.2.1

### PR DESCRIPTION
**Pull request by isanych**
freeglut on master is not building with gcc 10, updated to latest version, fixed gcc 10 issues.

### CODE REVIEW

File: ports/freeglut/portfile.cmake [hunk 1]
```diff
@@ -1,9 +1,9 @@
 include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/freeglut/portfile.cmake:1]:
```suggestion
```

File: ports/freeglut/portfile.cmake [hunk 2]
```diff
@@ -12,6 +12,7 @@ vcpkg_extract_source_archive_ex(
   PATCHES
     use_targets_to_export_x11_dependency.patch
     macOS_Xquartz.patch
+    gcc10.patch
 )
 
 if(VCPKG_CMAKE_SYSTEM_NAME AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
```
Comment (NancyLi1013) [inline ports/freeglut/portfile.cmake:18]:
```suggestion
if(NOT VCPKG_TARGET_IS_WINDOWS)
```

File: ports/freeglut/portfile.cmake [hunk 3]
```diff
@@ -45,6 +46,7 @@ vcpkg_configure_cmake(
 )
 
 vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/FreeGLUT)
 
 # Rename static lib (otherwise it's incompatible with FindGLUT.cmake)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
Comment (NancyLi1013) [inline ports/freeglut/portfile.cmake:52]: Could you please update this
` if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")`
as
` if(VCPKG_TARGET_IS_WINDOWS)`
?

Comment (NancyLi1013) [reply ports/freeglut/portfile.cmake:52]: Could you please update this
```
file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/freeglut)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/freeglut/COPYING ${CURRENT_PACKAGES_DIR}/share/freeglut/copyright)
```
as
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright) `?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by isanych** [comment]
@NancyLi1013 done

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for this PR @isanych .

**Comment by strega-nil** [comment]
@isanych what's up with the `glut.h` file? Shouldn't that be shipped with glut?

**Comment by isanych** [comment]
@strega-nil you right, it is still there, just has to be enabled

**Comment by strega-nil** [comment]
@isanych do you know what's wrong with x64_windows_static here? looks like a file rename failed with file not found?

**Comment by isanych** [comment]
@strega-nil with FREEGLUT_REPLACE_GLUT set both static and dynamic versions are called just glut and renaming is not needed anymore. vcpkg hopefully will be ok now, but to properly test unknown upstream changes it would be good to have small test project for each port which uses produced library - and if CI would build and run these binaries, it would give more confidence that changes are not breaking much.

For now it would be nice to get a notification with results when CI finish building.

**Comment by isanych** [comment]
looking at glui

**Comment by strega-nil** [comment]
Cool, thanks so much @isanych :)

**Pull request merged**